core/domain/skill_services.py

```diff
@@ -1147,3 +1147,83 @@ def filter_skills_by_mastery(user_id, skill_ids):
         if skill_id in filtered_skill_ids:
             arranged_filtered_skill_ids.append(skill_id)
     return arranged_filtered_skill_ids
+
+
+def get_untriaged_skill_summaries(
+    skill_summaries, skill_ids_assigned_to_some_topic, merged_skill_ids):
+    """Returns a list of skill summaries for all skills that are untriaged.
+
+    Args:
+        skill_summaries: list(SkillSummary). The list of all skill summary
+            domain objects.
+        skill_ids_assigned_to_some_topic: set(str). The set of skill ids which
+            are assigned to some topic.
+        merged_skill_ids: list(str). List of skill IDs of merged skills.
+
+    Returns:
+        list(SkillSummary). A list of skill summaries for all skills that
+        are untriaged.
+    """
+    untriaged_skill_summaries = []
+
+    for skill_summary in skill_summaries:
+        skill_id = skill_summary.id
+        if (skill_id not in skill_ids_assigned_to_some_topic) and (
+                skill_id not in merged_skill_ids):
+            untriaged_skill_summaries.append(skill_summary)
+
+    return untriaged_skill_summaries
+
+
+def get_categorized_skill_ids_and_descriptions():
+    """Returns a CategorizedSkills domain object for all the skills that are
+    categorized.
+
+    Returns:
+        CategorizedSkills. An instance of the CategorizedSkills domain object
+        for all the skills that are categorized.
+    """
+    topics = topic_fetchers.get_all_topics()
+
+    categorized_skills = skill_domain.CategorizedSkills()
+
+    skill_ids_and_details = []
+    skill_ids = []
+
+    for topic in topics:
+        subtopics = topic.subtopics
+        subtopic_titles = [subtopic.title for subtopic in subtopics]
+        categorized_skills.add_topic(topic.name, subtopic_titles)
+
+        for skill_id in topic.uncategorized_skill_ids:
+            skill_ids.append(skill_id)
+            skill_ids_and_details.append({
```

Comment:
I don't really understand why you need skill_ids_and_details. Why not just:

- Iterate to get all the skill IDs
- Fetch the skill descriptions
- Do one more iteration and update the categorized skills domain object directly, now that you already have the skill IDs to description map.

Reply:
We need skill_ids_and_details in order to store the information about the topic name and subtopic title the skill belongs to. This information is essential to update the domain object.

Reply:
I know that; but couldn't you add them while iterating in the step for the third bullet-point I mentioned above?